### DISCUSSION TITLE
What's the best workflow for the dev/test loop?

### DISCUSSION THREAD

**Discussion by giant-teapot** [fvtt-fria-ligan/forbidden-lands-foundry-vtt#25] [Q&A] [answered]
This is a very dumb question, one that I probably should have looked into a while ago.

Being new at Foundry, and not exactly literate regarding JS development, I'm not sure what is the "best practice" regarding how to test a change made on the system.

Here is my own dirty and clunky workflow:
- I have the project git repo somewhere
- I create a symbolic link from the repo to the `systems` directory
- I edit code, then head to an opened game and hit F5

Apparently, I cannot use a dev system "next" to the stable one without editing `system.json` which is… unfortunate.

How do you guys do it? Any productivity tips for the masses?

**Comment by jimorie** [comment]
I do the same as you. Following. :)

**Comment by aMediocreDad** [comment, marked as answer]
> Apparently, I cannot use a dev system "next" to the stable one without editing `system.json` which is… unfortunate.

Not sure what you want to get out of that, but you can open multiple instances of Foundry. I have one instance of Foundry open in vscode using a launch config, it is pointing towards a userdata folder that is different to my production data folder. If I want to compare I can always just open my production foundry instance and compare/test.

Multiple workflow resources here: https://foundryvtt.wiki/en/development/resources

Personally I have a setup similar to DJ Addis setup, along with a pure nodejs setup for when I am mobile.

**Reply by giant-teapot** [reply]
> Not sure what you want to get out of that, but you can open multiple instances of Foundry. I have one instance of Foundry open in vscode using a launch config, it is pointing towards a userdata folder that is different to my production data folder. […]

I can't believe I missed the `--dataPath` option. Turns out I never stumbled upon this article:
https://foundryvtt.com/article/configuration/

`--dataPath` will address most of what I was annoyed with. Sorry it wasn't clear in the original post, but what I want is pretty simple and downright boring:

- I want a dev system that is not overriding the stable system.
- That way, even if I break things in my local repo, the stable system stays unaffected and my games won't have as many glitches.
- It's pretty obvious how to do, once you know that `--dataPath` exists. 😑

I take another look at some of the articles in the wiki page you've linked.
Still need to work on my workflow, but I've got somewhere to start.

Meanwhile, don't hesitate to share any productivity tips. 😄
